token/js/test/e2e-2022/tokenmeta.test.ts

```diff
@@ -0,0 +1,86 @@
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import type { Connection, PublicKey, Signer } from '@solana/web3.js';
+import { sendAndConfirmTransaction, Keypair, SystemProgram, Transaction } from '@solana/web3.js';
+import {
+    createMint,
+    createInitializeAccountInstruction,
+    getAccountLen,
+    ExtensionType,
+    initializeMetadata,
+} from '../../src';
+import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection } from '../common';
+import { MINT_SIZE } from '../../src/state/mint';
+
+const TEST_TOKEN_DECIMALS = 2;
+const EXTENSIONS = [ExtensionType.CpiGuard];
+
+describe.only('meta', () => {
+    let connection: Connection;
+    let payer: Signer;
+    let owner: Keypair;
+    let account: PublicKey;
+    let mintKeypair: Keypair;
+    let mintAuthority: Keypair;
+
+    before(async () => {
+        connection = await getConnection();
+        payer = await newAccountWithLamports(connection, 1000000000);
+        owner = Keypair.generate();
+    });
+
+    beforeEach(async () => {
+        mintKeypair = Keypair.generate();
+        mintAuthority = Keypair.generate();
+        const accountKeypair = Keypair.generate();
+        account = accountKeypair.publicKey;
+        let accountLen = getAccountLen(EXTENSIONS);
+        accountLen = 600;
+        const lamports = await connection.getMinimumBalanceForRentExemption(accountLen);
+
+        const mint = await createMint(
+            connection,
+            payer,
+            mintAuthority.publicKey,
+            mintAuthority.publicKey,
+            TEST_TOKEN_DECIMALS,
+            mintKeypair,
+            undefined,
+            TEST_PROGRAM_ID
+        );
+
+        const transaction = new Transaction().add(
+            SystemProgram.createAccount({
+                fromPubkey: payer.publicKey,
+                newAccountPubkey: account,
+                space: accountLen,
+                lamports,
+                programId: TEST_PROGRAM_ID,
+            }),
+            createInitializeAccountInstruction(account, mint, owner.publicKey, TEST_PROGRAM_ID)
+        );
+
+        await sendAndConfirmTransaction(connection, transaction, [payer, accountKeypair], undefined);
+    });
+
+    it('Can initialize token metadata', async () => {
```

Comment:
Currently throwing this error:
```Error: failed to send transaction : Transaction simulation failed: Error processing Instruction 0: custom prog ram error: 0x33```

Think it maps to `Calculated fee does not match expected fee`
https://github.com/solana-labs/solana-program-library/blob/master/token/program-2022/src/error.rs#L126C14-L126C56

Reply:
Yeah, i'm quite sure i'm not setting up the accounts correctly - just can't figure out the right way